### PR TITLE
fix(share/getters): set minTimeout

### PR DESCRIPTION
**Pull request by vgonkivs**
Set minTimeout in case the context hasn't contained it.

### CODE REVIEW

File: share/getters/cascade.go [hunk 1]
```diff
@@ -119,13 +120,21 @@ func cascadeGetters[V any](
 		}
 	}()
 
+	minTimeout := time.Duration(0)
+	_, ok := ctx.Deadline()
+	if !ok {
+		// in this case minTimeout will be applied for all getters,so each of them
+		// will have 1 minute timeout.
+		minTimeout = time.Minute
+	}
```
Comment (walldiss) [inline share/getters/cascade.go:129]: Well, we still may encounter case, when user set 1Hour timeout here.
Perhaphs limiting timeout to be strict max 3min would be simple and robust. 
```
ctx = WithTimeout(ctx,3min)
```

**Comment by renaynay** [comment]
several integration tests failing @vgonkivs

**Comment by vgonkivs** [comment]
I'm not sure if my changes caused that.

File: share/getters/cascade.go (hunk 1 above)
Comment (vgonkivs) [reply share/getters/cascade.go:129]: > Well, we still may encounter case, when user set 1Hour timeout here. 

True, but then the user explicitly wants to wait 1h. I wanted to cover the case where he hasn't specified anything at all.

**Comment by cristaloleg** [comment]
Looks like 1min timeout is just too much:

> panic: test timed out after 10m0s

Maybe in tests we should keep 15sec or 30sec timeout, 2c.

**Review by Wondertan — APPROVED** [review]

**Review by cristaloleg — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
utack

**Pull request merged**